### PR TITLE
ENH: initial `_array_api` module + for `dpnp`/`dpctl` extension

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
Please do not merge/review yet.

# TODO
* remove utils._namespace module and move some funcs from there to _array_api
* check for dpnp/dpctl and branches where it is required
* move dpnp_converter to array_api module

### CODE REVIEW

File: sklearnex/utils/_array_api.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    device : str
+        The name of the device on which to allocate the arrays. Can be None to
+        indicate that the default value should be used.
+
+    dtype_name : str
+        The name of the data type to use for arrays. Can be None to indicate
+        that the default value should be used.
+    """
+    for array_namespace in [
```
Comment (samir-nasibli) [inline sklearnex/utils/_array_api.py:76]: Will be removed. Just for check. Will leave only DPNP/DPCTL related here.

**Comment by samir-nasibli** [comment]
This branch is almost outdated. Will be closed.
Some changes from here will be moved to the seperate PRs
Currently DBSCAN related work on #2014

**Comment by icfaust** [comment]
@samir-nasibli should this be closed?

**Comment by samir-nasibli** [comment]
> @samir-nasibli should this be closed?

https://github.com/intel/scikit-learn-intelex/pull/1813#issuecomment-2311895973
This will be closed and the branch deleted.

**Pull request closed without merging**